src/coreclr/vm/method.hpp

```diff
@@ -55,17 +55,27 @@ EXTERN_C VOID STDCALL NDirectImportThunk();
 #define METHOD_TOKEN_RANGE_BIT_COUNT (24 - METHOD_TOKEN_REMAINDER_BIT_COUNT)
 #define METHOD_TOKEN_RANGE_MASK ((1 << METHOD_TOKEN_RANGE_BIT_COUNT) - 1)
 
-enum class AsyncMethodType
+enum class AsyncMethodKind
```

Comment:
Just curious, but why do you prefer "kind" over "type"?

Reply:
I think inside the type system implementation referring to someting as "type" when it is not a type adds extra confusion. 
"Kind" seems slightly better, especially when we end up with fields/locals called "type".